src/main/scala/wolfendale/nunjucks/TemplateNode.scala

```diff
@@ -94,6 +96,40 @@ object TemplateNode {
     final case class ConditionalContent(condition: expression.syntax.AST, content: Partial)
   }
 
+
+  final case class Switch(exprToMatch:expression.syntax.AST.Expr, contents: Seq[Switch.ConditionalContent], default: Option[Partial]) extends Tag {
+
+    override def eval: State[Context, String] = {
+
+      val all: Seq[Switch.ConditionalContent] = default
+        .map(contents :+ Switch.ConditionalContent(Seq(exprToMatch), _))
+        .getOrElse(contents)
+
+      val collapsedConditionals = all.foldRight(Seq.empty[Switch.ConditionalContent]){
+        (m, n) => if(m.content.isEmpty){
+          val last = n.head
+          Switch.ConditionalContent(last.condition ++ m.condition, last.content) +: n.tail
+        }else {
+          m +: n
+        }
+      }
+
+      collapsedConditionals.foldLeft(State.pure[Context, Option[String]](None)) { (m, n) =>
+        m.flatMap {
+          case None =>
+            Monad[State[Context, *]].ifM(State.inspect(context => n.condition.exists(_.eval(context).equals(exprToMatch.eval(context)))))(
+              ifTrue = n.content.eval.map(_.some),
+              ifFalse = State.pure(None))
+          case some => State.pure(some)
+        }
+      }
+      }.map(_.getOrElse(""))
+  }
+
+  object Switch {
+    final case class ConditionalContent(condition: Seq[expression.syntax.AST], content: Partial)
```

Comment:
Deos `condition` need to be a `Seq`? it looks like where it's being created it's only ever going to have one element.

Reply:
it's there because when the case is supposed to fall through to the next one I stack them up in the sequence. I could turn that into a tuple in the fold I suppose instead of having the template construct that